### PR TITLE
DS-267 | @rebeccahongsf | fetch payment ref id and correctly setup memberships mock

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- fetch payment ref id from mock membership
- correctly setup memberships mock
**Note:** Widget might render without data due to known GiveGab bug.

# Review By (Date)
- When possible

# Review Tasks

### Test for with Payment Id
1. Checkout branch on your local
2. Enable mockserver by adding `MEGAPROFILE_MOCK='true'` to your .env
3. Navigate to `/membership/payment-history`
4. Verify that the widget appears in a card, similar to the Membership form
5. Navigate to `/membership/payment-update`
8. Verify that the widget appears in a card, similar to the Membership form
9. Review code

### Test for No Payment Id
1. Navigate to https://deploy-preview-746--stanford-alumni.netlify.app/membership/payment-history
2. Verify that a payment id error appears
3. Navigate to https://deploy-preview-746--stanford-alumni.netlify.app/membership/payment-update
4. Verify that a payment id error appears
5. Review code

# Associated Issues and/or People
- DS-267

### CODE REVIEW

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 1]
```diff
@@ -129,11 +141,31 @@ const LightFormPage = (props) => {
...
+                      <p className="su-ml-03em">Loading form...</p>
+                      <noscript>
+                        Sorry, but you must have Javascript enabled to use the
+                        form.
+                      </noscript>
+                    </div>
+                  ) : (
+                    <>
+                      {error ? (
+                        <div>
```
Comment (rebeccahongsf) [inline src/components/page-types/lightFormPage/lightFormPage.js:156]: Additional error message handling will be implemented in [DS-433](https://stanford.atlassian.net/browse/DS-433)

File: src/api/auth/profile.js [hunk 2]
```diff
@@ -52,6 +55,7 @@ const storyblokPreviewPassthrough = async (req, res, next) => {
   const isEditor = await isStoryblokEditor(req);
   if (isEditor) {
     res.json(fullprofileMockData);
+    res.json(membershipsMockData);
```
Comment (sherakama) [inline src/api/auth/profile.js:58]: Two sets of JSON? Is this allowed?

**Comment by sherakama** [comment]
Perhaps add a link to the help SNOW form in the error message

**Comment by rebeccahongsf** [comment]
> Perhaps add a link to the help SNOW form in the error message

Additional error message handling will be implemented in [DS-433](https://stanford.atlassian.net/browse/DS-433). Should I still add a SNOW form link in the meantime? 🤔

File: src/api/auth/profile.js (hunk 2 above)
Comment (rebeccahongsf) [reply src/api/auth/profile.js:58]: You are 100% correct 😅 res can only accept one response of JSON. It should actually be

```
res.json({...fullprofileMockData, ...membershipMockData})
```

I'm on the fix! 🫡

**Review by sherakama — APPROVED** [review]
Okie doke.

**Pull request merged**